packages/client/test/end-to-end/publisher-message-validation.test.ts

```diff
@@ -0,0 +1,46 @@
+import { StreamrClient } from '../../src'
+import { createTestStream, fetchPrivateKeyWithGas, getCreateClient } from '../test-utils/utils'
+import { fastPrivateKey, wait } from 'streamr-test-utils'
+import { StreamID } from 'streamr-client-protocol'
+
+const createClient = getCreateClient()
+
+const TIMEOUT = 20 * 1000
+
+const PROPAGATION_WAIT_TIME = 2000
+
+describe('publisher message validation (NET-773)', () => {
+    let publisherClient: StreamrClient
+    let subscriberClient: StreamrClient
+    let streamId: StreamID
+
+    beforeEach(async () => {
+        publisherClient = await createClient({
+            auth: {
+                privateKey: fastPrivateKey()
+            }
+        })
+        subscriberClient = await createClient({
+            auth: {
+                privateKey: await fetchPrivateKeyWithGas()
+            }
+        })
+        const stream = await createTestStream(subscriberClient, module)
+        streamId = stream.id
+    }, TIMEOUT)
+
+    it('publishing message with insufficient permissions prevents message from getting sent to network', async () => {
+        let subscriberReceivedMsgs = 0
+        const subscription = await subscriberClient.subscribe(streamId, () => {
```

Comment:
How about if there were two jest mock functions:
```
onMessage = jest.fn()
onError = jest.fn()
```

and in expect line we check that neither is called:
```
expect(onMessage).not.toHaveBeenCalled()
expect(onError).not.toHaveBeenCalled()
```